### PR TITLE
fix: the ebpf program in the tcpretrans plugin failed to load

### PR DESCRIPTION
**Pull request by wenhuwang**
# Description
Fix the issues of abnormal status of retina-agent caused by opening tcpretrans plugin. The retina-agent nil pointer is because the [Tracer.socketEnricherMap](https://github.com/inspektor-gadget/inspektor-gadget/blob/main/pkg/gadgets/trace/tcpretrans/tracer/tracer.go#L41) field of the tcpretrans plugin is nil.

## Related Issue

#311 

## Checklist

- [x] I have read the [contributing documentation](https://retina.sh/docs/contributing).
- [x] I signed and signed-off the commits (`git commit -S -s ...`). See [this documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification) on signing commits.
- [x] I have correctly attributed the author(s) of the code.
- [x] I have tested the changes locally.
- [ ] I have followed the project's style guidelines.
- [ ] I have updated the documentation, if necessary.
- [ ] I have added tests, if applicable.

---

Please refer to the [CONTRIBUTING.md](../CONTRIBUTING.md) file for more information on how to contribute to this project.

### CODE REVIEW

**Comment by rbtr** [comment]
@wenhuwang this doesn't seem to fully resolve the problem. Instead of a nil pointer, now there is an error instantiating the socketEnricher:
> failed to reconcile plugin tcpretrans: failed to new socketEnricher: read BPF iterator: checking if current pid namespace is host pid namespace: host.Init() must be called before calling isHostNamespace()

**Review by rbtr — COMMENTED** [review]
needs to be tested

**Review by rbtr — CHANGES_REQUESTED** [review]
needs to be tested

**Comment by wenhuwang** [comment]
> @wenhuwang this doesn't seem to fully resolve the problem. Instead of a nil pointer, now there is an error instantiating the socketEnricher:
> 
> > failed to reconcile plugin tcpretrans: failed to new socketEnricher: read BPF iterator: checking if current pid namespace is host pid namespace: host.Init() must be called before calling isHostNamespace()

@rbtr Sorry, I'm on vacation recently. I have tested it locally before. I will verify this problem in a few days.

**Comment by wenhuwang** [comment]
I have not reproduced this problem in my local environment. 
> failed to reconcile plugin tcpretrans: failed to new socketEnricher: read BPF iterator: checking if current pid namespace is host pid namespace: host.Init() must be called before calling isHostNamespace()

Platform (please complete the following information):
OS: Ubuntu 18.04.5 LTS
Kubernetes Version: 1.22.2
Host: self-host
Kernel Version: 5.10.87-051087-generic
Retina Version: v0.0.8

Do you have any specific steps to reproduce the problem? @rbtr

File: pkg/plugin/tcpretrans/tcpretrans_linux.go [hunk 1]
```diff
@@ -56,6 +58,12 @@ func (t *tcpretrans) Init() error {
 	// Create tracer. In this case no parameters are passed.
 	t.tracer = &tracer.Tracer{}
```
Comment (anubhabMajumdar) [inline pkg/plugin/tcpretrans/tcpretrans_linux.go:59]: I believe IG also requires `host.Init()` call before tracer is initialized; similar to https://github.com/microsoft/retina/blob/0dbffcbf9726710702121f71553d0af397d250b2/pkg/plugin/dns/dns_linux.go#L55 . Can you add that as well?

**Review by anubhabMajumdar — COMMENTED** [review]
Thanks for the change. Can you add details as to how you verified this change?

**Comment by wenhuwang** [comment]
> Thanks for the change. Can you add details as to how you verified this change?
1. Before adding this change, the retina-agent would panic because of the nil pointer.
2. After adding this change, retina-agent runs normally and can get tcpretrans plugin metrics.
3. When troubleshooting this problem through the logs, I can only see the "Initialized tcpretrans plugin" log but not the "failed to new socketEnricher" related logs.

> failed to reconcile plugin tcpretrans: failed to new socketEnricher: read BPF iterator: checking if current pid namespace is host pid namespace: host.Init() must be called before calling isHostNamespace()

File: pkg/plugin/tcpretrans/tcpretrans_linux.go (hunk 1 above)
Comment (wenhuwang) [reply pkg/plugin/tcpretrans/tcpretrans_linux.go:59]: OK, that's what i thought

**Review by rbtr — APPROVED** [review]
lgtm, thanks @wenhuwang

**Pull request merged**